### ISSUE TITLE
Generate the first model based on the first image batch.

### ISSUE THREAD

**Issue by GreNait** [GreNait/technical-thesis#4]
- [x] include filter to decrease image size to 150x113 size
- [x] train first model

**Comment by GreNait** [comment]
TF workflow guide for image classification with cats and dogs:

https://www.tensorflow.org/tutorials/images/classification

**Comment by GreNait** [comment]
https://www.tensorflow.org/tutorials/customization/custom_training_walkthrough 
-> looks like a better walkthrough, because I do not have my data yet organzied.

-> first part was similoar to the first walktrhough and not helpful. Because i couldn't use tf.keras.utils.get_file <- download data, which I am not doing

**Comment by GreNait** [comment]
Using the youtube videos from sentdex looks promising

https://www.youtube.com/watch?v=j-3vuBynnOE

**Comment by GreNait** [comment]
Just found out, that using the "#&&" in vscode, defines part of the program as "cells" like in jupyter notebooks. This is great, for coding and testing, bigger data amounts do not need to be reloaded with every start of the program.

**Comment by GreNait** [comment]
Following along the tutorial from sentdex adding now convolutions etc.

https://www.youtube.com/watch?v=WvoLTXIjBYU&list=PLQVvvaa0QuDfhTox0AjmQ6tvTgMBZBEXN&index=4&t=0s

**Comment by GreNait** [comment]
First model trained, accuracy is really bad with 34% BUT it worked !!

![image](https://app.gitkraken.com/api/glo/boards/5e43bac3757b7a00112ba9d7/attachments/5e48158f757b7a00112c3fe1)

**Comment by GreNait** [comment]
The nvidia-smi command somehow worked today and the model was a lot faster too calculate

![image](https://app.gitkraken.com/api/glo/boards/5e43bac3757b7a00112ba9d7/attachments/5e49149e184ba700103018ef)

**Comment by GreNait** [comment]
Still, the accuracy of the model with 100 epochs is miserable
![image](https://app.gitkraken.com/api/glo/boards/5e43bac3757b7a00112ba9d7/attachments/5e4914e1a326a20010898073)

**Comment by GreNait** [comment]
Using the tensorboard for showing the training of the model

https://www.youtube.com/watch?v=BqgTU7_cBnk&list=PLQVvvaa0QuDfhTox0AjmQ6tvTgMBZBEXN&index=5&t=24s

![image](https://app.gitkraken.com/api/glo/boards/5e43bac3757b7a00112ba9d7/attachments/5e491a050e8c720011033f84)

**Comment by GreNait** [comment]
I just had an "blackout" of the system during the training of the model. The fan of the GPU started to spin brutaly and the display went off. I had to hard restart the pc again ... Some googling didn't help yet.

**Comment by GreNait** [comment]
After updating the gpu driver via the ubunut system driver setting and rebooting the system, the new model was trained witch a lot higher neurons and 100 epochs. Still, the accuracy is terrible with 35%, but at least it looks like it is working.

**Comment by GreNait** [comment]
The black out of the display happend again. I assume, that this might be caused by the memory allocation of the gpu by tensorflow. I searched for a way to limit the possible allocation. There was an blogpost here:
https://github.com/tensorflow/tensorflow/issues/25138

Recommending following two lines:

tf.config.gpu.set_per_process_memory_fraction(0.95)
tf.config.gpu.set_per_process_memory_growth(True)

I decided, to start with 95% allocation.

**Comment by GreNait** [comment]
Didn't work. Had another blakc out ...
